### DISCUSSION TITLE
Columnchart - Unable to change font of text in Axis

### DISCUSSION THREAD

**Discussion by f3rn0s** [cetz-package/cetz#506] [Q&A] [answered]
Okay so, by my understanding the default way the text in the axis gets rendeered is:

```
$#value$
```

Now, by specifying a custom format function for the axis I should be able to get it to render as normal text, and thus have the same font as the rest of my document. The problem comes in when we look at the implementation of bar/column charts. These define the axis themselves:

```
  let x = axes.axis(min: min-value, max: max-value,
                    label: x-label,
                    ticks: (grid: true, step: x-tick-step,
                            minor-step: none,
                            unit: x-unit, decimals: 1,
                            list: x-ticks))
  let y = axes.axis(min: data.len(), max: -1,
                    label: y-label,
                    ticks: (grid: true,
                            step: none,
                            minor-step: none,
                            list: y-tic-list))
 ```
 
From my understanding, since I can't edit these function calls to supply my own format function to them, I can't actually make it so the font renders as normal content.

In this case, this leads to a weird looking chart:

![image](https://github.com/johannes-wolf/cetz/assets/1351279/092781b8-7f47-4bf1-85c1-7a06c694caa0)

If ther eis a way to fix this and I've missed it, my apologies.

**Comment by johannes-wolf** [comment]
You have to wait for `0.2.0` to get released (or, if you are using Typst locally, you can install the cetz 0.2.0 branch on your machine). With 0.2.0 it is: `x-format: label => text(<your-formatting..>, label)` - you can provide a custom label formatting callback.

**Comment by f3rn0s** [comment]
Hey @johannes-wolf, with the push of 0.2.0 to latest, I'm still struggling to get this working.

You mention x-format, but in this case it would have to be something like y-format? I would love some help on this cause I can't seem to resolve the original issue using options mentioned in the manual.

**Comment by johannes-wolf** [comment]
You can give your data labels as content.
Example:
```typst
  let data = (
    ([Item A], 1),
    ([*Bold Text*], 4)
  )
  
  cetz.chart.columnchart(size: (8, 6),
    data,
    label-key: 0,
    value-key: 1,
  )
```

The `x-format` and `y-format` only work on auto-generated tick-labels.
You can also set the tick-labels using `x-ticks: (...)` (see manual).

**Comment by johannes-wolf** [comment]
Can you give example code and what you want to achieve?

**Comment by f3rn0s** [comment]
Sure!

This is the output:

![image](https://github.com/johannes-wolf/cetz/assets/1351279/a1bfdc36-cb01-42b0-ada6-241179b6f68a)

Here is the code used to generate it:

```typst
#let password_lengths = (
    "< 8" : 1,
    "8"   : 5,
    "9"   : 10,
    "10"  : 55,
    "11"  : 1337,
    "12"  : 100,
    "13"  : 43,
    "> 13": 12,
)

#align(center, canvas({
    import draw: *

    let colors = palette.new(colors: (color_palette.at("main").lighten(30%), black))

    // Remove background grid
    set-style(
        grid: (
            stroke: none,
        ),
    )

    chart.columnchart(
        size: (auto, password_lengths.keys().len()),
        mode: "stacked",
        value-key: (1,),
        bar-style: colors,
        {
            let result_data = ()

            for key in password_lengths.keys() {
                result_data.push((key, password_lengths.at(key)))
            }

            //panic(result_data)
           result_data
        },
    )
}))
```

Since the y-tick is auto-generated, I'd like it to be `text` as opposed to `$$`, but I wasn't sure how to get that working.

**Comment by johannes-wolf** [comment, marked as answer]
Set `y-format: v => [#calc.round(v)],`

**Comment by f3rn0s** [comment]
That worked perfectly, thank you very much :)
